packages/vite-plugin-cloudflare/playground/vitest-setup.ts

```diff
@@ -181,7 +181,11 @@ beforeAll(async (s) => {
 }, 15_000);
 
 beforeEach(async () => {
-	await page.goto(viteTestUrl);
+	try {
+		await page.goto(viteTestUrl);
+	} catch (error) {
+		// Expected to error for middleware-mode test as `viteTestUrl` will not be valid
+	}
```

Comment:
could we instead avoid running the `page.goto` in the first place for the middleware-mode app?

Reply:
Is there a way to opt out of the `beforeEach` in a single test file?

Reply:
I was thinking of something along these lines:
```js
beforeEach(async () => {
	if (rootDir.includes("middleware-mode")) {
	    // there is no `ivteTestUrl` to navigate to for middleware-mode
		return;
	}
	
	await page.goto(viteTestUrl);
});
```

it is a bit hardcoded but it might be better than swallowing all potential exceptions here? 🤔

Reply:
Ah, nice idea. I'll try that.

Reply:
See https://github.com/cloudflare/workers-sdk/pull/9108#discussion_r2069988230